src/auth/components/oidcCallback/OidcCallback.tsx

```diff
@@ -14,47 +17,47 @@ function OidcCallback({
 }: RouteChildrenProps): React.ReactElement | null {
   const { t } = useTranslation();
   const redirectToErrorPage = useErrorPageRedirect();
+  const generigErrorString = 'authentication.genericError.message';
 
-  useEffect(() => {
-    authService
-      .endLogin()
-      .then(() => {
-        history.replace('/', getLinkRedirectState());
-      })
-      .catch((error: Error) => {
-        // Handle error caused by device time being more than 5 minutes off
-        if (
-          error.message.includes('iat is in the future') ||
-          error.message.includes('exp is in the past')
-        ) {
-          redirectToErrorPage({
-            message: t('authentication.deviceTimeError.message'),
-          });
-        } else if (
-          // Handle error caused by end user choosing Deny in Tunnistamo's
-          // permission request
-          error.message ===
-          'The resource owner or authorization server denied the request'
-        ) {
-          redirectToErrorPage({
-            message: t('authentication.permissionRequestDenied.message'),
-          });
-        } else {
-          // Send other errors to Sentry for analysis
-          Sentry.captureException(error);
-          // Give user a generic error
-          redirectToErrorPage({
-            message: t('authentication.genericError.message'),
-          });
-        }
+  const onSuccess = () => {
+    // Successful login - redirect to profile page
+    history.replace('/', getLinkRedirectState());
+  };
+
+  const onError = (error?: OidcClientError) => {
+    if (!error) {
+      redirectToErrorPage({
+        message: t('authentication.genericError.message'),
```

Comment:
Generic message variable could be used here 🤔

Reply:
Fixed